shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/plants/data/FilePlantsDataSource.kt

```diff
@@ -1,17 +1,62 @@
 package com.vml.tutorial.plantshop.plants.data
 
+import com.vml.tutorial.plantshop.MR.files.plants
+import com.vml.tutorial.plantshop.PlantDatabase
+import com.vml.tutorial.plantshop.core.data.FileReader
+import com.vml.tutorial.plantshop.core.data.toPlant
 import com.vml.tutorial.plantshop.core.data.toPlants
 import com.vml.tutorial.plantshop.plants.domain.Plant
 import com.vml.tutorial.plantshop.plants.domain.PlantsDataSource
-import com.vml.tutorial.plantshop.MR.files.plants
-import com.vml.tutorial.plantshop.core.data.FileReader
-
 
 class FilePlantsDataSource(
```

Comment:
Its up to you but I think we can separate db, favorite, and the file source because they have different responsibilities. Later in repository, you can combine them.

So you can have a FilePlantsDataSource, and a LocalPlantsDataSource (db and fav) in the repository you can choose which data source do you want to use. All of them or a specific for a specific task... :) 

You can leave it as it is if you don't want to touch this again :D

Reply:
db, favorite, and the file sources are separated. I didn't separate the repository since they are dependent on each other but if you are unhappy with it, I can come with another solution.